### PR TITLE
Closes #18150: select tabs button not displayed

### LINKED ISSUE #18150: [Bug] "Select tabs" button not displayed
* STR:
open 2 or more websites in normal mode
switch to private browsing mode
open a website
close opened website
Go to normal tabs 
"Select tabs" button not displayed

* expected behaviour
show "Select tabs" button

* actual behaviour
"Select tabs" button not visible

reproducible:
* version: 86.1.1
* version: 87.0.0-beta.1 (Build #2015795065) AC: 73.0.0, 69a800e47 GV: 87.0-20210222154334 AS: 71.0.0
* version: Nightly 210224 (Build #2015795113) AC: 73.0.20210223143117, 05f3e3cc7 GV: 87.0a1-20210222084406 AS: 71.0.0

interesting note (continuation from STR):
go back to pbm
come back to normal bm
"Select tabs" button is visible

excuse cant submit logs.

> android 9, Galaxy J7 Nxt

### PR DESCRIPTION
**Pull request by VitalyVPinchuk**
Fixes #18150

`NotifyItemChanged()` with `payload` has no effect for the first time as the whole view needs to be updated.
Added a method to update private mode state in `SaveToCollectionsButtonAdapter` to address the issue.
Added smoke test but not sure if such a test is needed.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 1]
```diff
@@ -1055,6 +1055,33 @@ class SmokeTest {
...
+        }.openTabDrawer {
+        }.openNewTab {
+        }.submitQuery(secondWebPage.url.toString()) {
+            mDevice.waitForIdle()
+        }.openTabDrawer {
+        }.toggleToPrivateTabs {
+        }.openNewTab {
+        }.dismissSearchBar { }
+
+        homeScreen { }
```
Comment (Mugurell) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:1078]: nit: Maybe we can have the same formatting as the above here?
as in 

```suggestion
        homeScreen {
        }.openTabDrawer {

```

**Review by Mugurell — COMMENTED** [review]
Thank you for looking into this!
I really like the tests!

I see the issue stemming from the fact that when we decide whether to show or not the button there are two things checked:
- a fixed `isPrivate` value
- a dynamic ~`isModeNormal` value

I'd want to ask you how you'd feel about maybe refactoring that 
`private val isPrivate: Boolean = false` property to something like 
`private val isPrivate: () -> Boolean = { false }`
(that in TabTrayView would be initialized with `{ isPrivateModeSelected }`) ?

There is a lot of technical debt already so I agree there's no clean solution but I'd prefer to not add new APIs (the new setter) and rather force clients to initialize the adapter with `isPrivate` also as a dynamic property.

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 1 above)
Comment (VitalyVPinchuk) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:1078]: Hello @Mugurell 
I really appreciate your feedback and help! I'll fix all the remarks.

**Review by Mugurell — APPROVED** [review]
LGTM.
Thank you!

**Pull request merged**